modules/configuration-api/src/main/java/org/apache/ignite/configuration/notifications/ConfigurationNotificationEvent.java

```diff
@@ -35,11 +35,31 @@
      */
     @Nullable VIEWT oldValue();
 
+    /**
+     * Returns old value of the parent (any from the root) or current configuration.
+     *
+     * <p>For example, if we changed the child configuration, then we can get both the parent and the current child configuration.
+     *
+     * @param viewClass Configuration interface, for example {@code RootView.class}.
+     * @param <T> Configuration type.
+     */
+    @Nullable <T> T oldValue(Class<T> viewClass);
```

Comment:
In fact, `oldValue()` is a special case of this method, do you think it's worth getting rid of `oldValue()` ?

Reply:
We should leave the method without parameters, because unlike `oldValue(FooView.class)`, `oldValue()` may return `String` / `Integer` / etc.

Reply:
Ok